libs/server/Resp/BasicCommands.cs

```diff
@@ -726,44 +664,41 @@ private bool NetworkIncrement<TGarnetApi>(RespCommand cmd, ref TGarnetApi storag
                          cmd == RespCommand.DECR);
 
             var key = parseState.GetArgSliceByRef(0);
-            var sbKey = key.SpanByte;
 
-            ArgSlice input = default;
-            if (cmd == RespCommand.INCRBY || cmd == RespCommand.DECRBY)
+            var input = new RawStringInput
             {
-                // Parse value argument
-                // NOTE: Parse empty strings for better error messages through storageApi.Increment
-                var sbVal = parseState.GetArgSliceByRef(1).SpanByte;
-                var valPtr = sbVal.ToPointer() - RespInputHeader.Size;
-                var vSize = sbVal.Length + RespInputHeader.Size;
-                ((RespInputHeader*)valPtr)->cmd = cmd;
-                ((RespInputHeader*)valPtr)->flags = 0;
-                input = new ArgSlice(valPtr, vSize);
-            }
-            else if (cmd == RespCommand.INCR)
+                header = new RespInputHeader { cmd = cmd }
+            };
+
+            Span<byte> outputBuffer = stackalloc byte[NumUtils.MaximumFormatInt64Length + 1];
+            var output = ArgSlice.FromPinnedSpan(outputBuffer);
+
+            if (cmd == RespCommand.INCRBY || cmd == RespCommand.DECRBY)
             {
-                var vSize = RespInputHeader.Size + 1;
-                var valPtr = stackalloc byte[vSize];
-                ((RespInputHeader*)valPtr)->cmd = cmd;
-                ((RespInputHeader*)valPtr)->flags = 0;
-                *(valPtr + RespInputHeader.Size) = (byte)'1';
-                input = new ArgSlice(valPtr, vSize);
+                input.parseState = parseState;
+                input.parseStateStartIdx = 1;
+                storageApi.Increment(key, ref input, ref output);
             }
-            else if (cmd == RespCommand.DECR)
+            else if (cmd == RespCommand.INCR || cmd == RespCommand.DECR)
             {
-                var vSize = RespInputHeader.Size + 2;
-                var valPtr = stackalloc byte[vSize];
-                ((RespInputHeader*)valPtr)->cmd = cmd;
-                ((RespInputHeader*)valPtr)->flags = 0;
-                *(valPtr + RespInputHeader.Size) = (byte)'-';
-                *(valPtr + RespInputHeader.Size + 1) = (byte)'1';
-                input = new ArgSlice(valPtr, vSize);
-            }
+                var value = cmd == RespCommand.INCR ? "1" : "-1";
+                var valueBytes = Encoding.ASCII.GetBytes(value);
 
-            Span<byte> outputBuffer = stackalloc byte[NumUtils.MaximumFormatInt64Length + 1];
-            var output = ArgSlice.FromPinnedSpan(outputBuffer);
+                fixed (byte* ptr = valueBytes)
+                {
+                    // Prepare the parse state
+                    var valueSlice = new ArgSlice(ptr, valueBytes.Length);
+
+                    var tmpParseState = new SessionParseState();
```

Comment:
I think it is better to move the root into SessionParseState, seems unintuitive to read and maintain as it stands.
Also, why do we even need a new parse state here as this is within RespServerSession where we have a parse state we can reuse.